src/core/execute-serialize.c

```diff
@@ -0,0 +1,2133 @@
+                return r;
+
+        r = serialize_item(f, "exec-context-std-output", exec_output_to_string(c->std_output));
+        if (r < 0)
+                return r;
+
+        r = serialize_item(f, "exec-context-std-error", exec_output_to_string(c->std_error));
+        if (r < 0)
+                return r;
+
+        r = serialize_bool_elide(f, "exec-context-stdio-as-fds", c->stdio_as_fds);
+        if (r < 0)
+                return r;
+
+        if (c->std_input == EXEC_INPUT_NAMED_FD) {
+                r = serialize_item(f, "exec-context-std-input-fd-name", c->stdio_fdname[STDIN_FILENO]);
+                if (r < 0)
+                        return r;
+        }
+
+        if (c->std_output == EXEC_OUTPUT_NAMED_FD) {
+                r = serialize_item(f, "exec-context-std-output-fd-name", c->stdio_fdname[STDOUT_FILENO]);
+                if (r < 0)
+                        return r;
+        }
+
+        if (c->std_error == EXEC_OUTPUT_NAMED_FD) {
+                r = serialize_item(f, "exec-context-std-error-fd-name", c->stdio_fdname[STDERR_FILENO]);
+                if (r < 0)
+                        return r;
+        }
+
+        if (c->std_input == EXEC_INPUT_FILE) {
+                r = serialize_item(f, "exec-context-std-input-file", c->stdio_file[STDIN_FILENO]);
+                if (r < 0)
+                        return r;
+        }
+
+        if (c->std_output == EXEC_OUTPUT_FILE) {
+                r = serialize_item(f, "exec-context-std-output-file", c->stdio_file[STDOUT_FILENO]);
+                if (r < 0)
+                        return r;
+        }
+
+        if (c->std_output == EXEC_OUTPUT_FILE_APPEND) {
+                r = serialize_item(f, "exec-context-std-output-file-append", c->stdio_file[STDOUT_FILENO]);
+                if (r < 0)
+                        return r;
+        }
+
+        if (c->std_output == EXEC_OUTPUT_FILE_TRUNCATE) {
+                r = serialize_item(f, "exec-context-std-output-file-truncate", c->stdio_file[STDOUT_FILENO]);
+                if (r < 0)
+                        return r;
+        }
+
+        if (c->std_error == EXEC_OUTPUT_FILE) {
+                r = serialize_item(f, "exec-context-std-error-file", c->stdio_file[STDERR_FILENO]);
+                if (r < 0)
+                        return r;
+        }
+
+        if (c->std_error == EXEC_OUTPUT_FILE_APPEND) {
+                r = serialize_item(f, "exec-context-std-error-file-append", c->stdio_file[STDERR_FILENO]);
+                if (r < 0)
+                        return r;
+        }
+
+        if (c->std_error == EXEC_OUTPUT_FILE_TRUNCATE) {
+                r = serialize_item(f, "exec-context-std-error-file-truncate", c->stdio_file[STDERR_FILENO]);
+                if (r < 0)
+                        return r;
+        }
+
+        r = serialize_item_base64mem(f, "exec-context-stdin-data", c->stdin_data, c->stdin_data_size);
+        if (r < 0)
+                return r;
+
+        if (c->tty_path) {
```

Comment:
if tty_path is not set but you have StandardOutput=tty set, then we'll default to /etc/console, and treset it if needed. This this if block here should not be an if block